pydantic/dataclasses.py

```diff
@@ -38,6 +38,7 @@ def dataclass(
         validate_on_init: bool | None = None,
         kw_only: bool = ...,
     ) -> Callable[[type[_T]], type[PydanticDataclass]]:  # type: ignore
+        """Overload for `dataclass`."""
```

Comment:
Is it necessary to document the overloads? I don't think even the docstring linters care about it. 🤔

Reply:
Interrogate complains if it's not documented, and interrogate is useful for tracking coverage clearly. Ruff complains if it is documented. Noted for future research.